packages/rpc/src/encoding/basic.ts

```diff
@@ -209,13 +205,15 @@ export function createBasicEncoder(api: EncodingStrategyApi): EncodingStrategy {
         return proxy as any;
       }
 
-      return Object.keys(value).reduce(
-        (object, key) => ({
-          ...object,
-          [key]: decode((value as any)[key], retainedBy),
-        }),
-        {},
-      ) as any;
+      if (isBasicObject(value)) {
```

Comment:
If it's not basic, it'll be returned as is further down

Reply:
Not sure if there's "non basic" cases, where we'd need to iterate over properties.